router/router.go

```diff
@@ -1054,14 +1054,14 @@ func (worker *workerT) postStatusOnResponseQ(respStatusCode int, respBody string
 		if worker.rt.guaranteeUserEventOrder {
 			if addToFailedMap {
 				//#JobOrder (see other #JobOrder comment)
-				worker.failedJobIDMutex.RLock()
```

Comment:
IMO there is practically no way for a race to happen, since the only one who can write it is the go-routine itself.

Reply:
Even if `postStatusOnResponseQ` doesn't get invoked concurrently the usage of the mutex there is wrong and should be fixed, also because it adds unnecessary risks. If someone were to add an additional routine that writes there it would surely break. Do you think it's best not to merge this change?

Reply:
Please keep in mind that the goal is to move away from the current implementation
#2329 fixes various practical issues with event ordering with regards to the current implementation and introduces a new algorithm/approach. Although the new approach won't be used immediately, the plan is to replace the old one in due time.

That being said, I will introduce your changes to #2329 in order to avoid unnecessary merging conflicts :)

Reply:
Those changes are on #2329, @fracasula are you cool closing this?